db/auto_roll_logger.cc

```diff
@@ -32,8 +32,17 @@ Status AutoRollLogger::ResetLogger() {
 }
 
 void AutoRollLogger::RollLogFile() {
-  std::string old_fname = OldInfoLogFileName(
-      dbname_, env_->NowMicros(), db_absolute_path_, db_log_dir_);
+  uint64_t now = env_->NowMicros();
+  std::string old_fname;
+  // Try to check target name only 10 times at most
+  for (int i = 0; i < 10; i++) {
```

Comment:
Please, explain and comment this change. My understanding this all happens inline with a logging request and we do not want those to be expensive.